tests/output/test_printing.py

```diff
@@ -85,18 +104,134 @@ def test_print_db_cursor_json(context_mock, capsys):
     ]
 
 
-@mock.patch("snowcli.output.printing.click.get_current_context")
-def test_print_db_cursor_filters_columns(context_mock, capsys):
-    context = Context(Command("foo"))
-    context.params = {"output_format": "JSON"}
-    context_mock.return_value = context
+def test_print_multi_db_cursor_json(capsys):
+    mock_cursor = _create_mock_cursor()
+
+    output_data = OutputData(OutputFormat.JSON)
+    output_data.add_cursor(mock_cursor)
+    output_data.add_cursor(mock_cursor)
+    print_output(output_data)
+
+    assert _get_output_as_json(capsys) == [
+        [
+            {
+                "string": "string",
+                "number": 42,
+                "array": ["array"],
+                "object": {"k": "object"},
+                "date": "2022-03-21T00:00:00",
+            },
+            {
+                "string": "string",
+                "number": 43,
+                "array": ["array"],
+                "object": {"k": "object"},
+                "date": "2022-03-21T00:00:00",
+            },
+        ],
+        [
+            {
+                "string": "string",
+                "number": 42,
+                "array": ["array"],
+                "object": {"k": "object"},
+                "date": "2022-03-21T00:00:00",
+            },
+            {
+                "string": "string",
+                "number": 43,
+                "array": ["array"],
+                "object": {"k": "object"},
+                "date": "2022-03-21T00:00:00",
+            },
+        ],
+    ]
 
-    cur = _MockCursor()
-    print_db_cursor(cur, columns=["string", "object"])
 
-    captured = capsys.readouterr()
-    json_output = json.loads(captured.out)
-    assert json_output == [
+def test_print_different_data_sources_json(capsys):
+    output_data = OutputData(OutputFormat.JSON)
+    output_data.add_cursor(_create_mock_cursor())
+    output_data.add_string_output("Command done")
+    output_data.add_data([{"key": "value"}])
+
+    print_output(output_data)
+
+    assert _get_output_as_json(capsys) == [
+        [
+            {
+                "string": "string",
+                "number": 42,
+                "array": ["array"],
+                "object": {"k": "object"},
+                "date": "2022-03-21T00:00:00",
+            },
+            {
+                "string": "string",
+                "number": 43,
+                "array": ["array"],
+                "object": {"k": "object"},
+                "date": "2022-03-21T00:00:00",
+            },
+        ],
+        [{"result": "Command done"}],
```

Comment:
Should this be array? We know it's a single result.

Reply:
Right, changed